rubix-spi/src/main/java/com/qubole/rubix/spi/CacheUtil.java

```diff
@@ -208,7 +208,11 @@ private static void createCacheDirectory(String cacheDirPath)
   {
     final File cacheDir = new File(cacheDirPath);
     cacheDir.mkdirs();
-    cacheDir.setWritable(true, false);
+    File parentFile = cacheDir;
+    do {
+      parentFile.setWritable(true, false);
+      parentFile = parentFile.getParentFile();
+    } while (parentFile.getAbsolutePath().contains("fcache"));
```

Comment:
Remove "fcache" and get it from the config

Reply:
Made the change